src/odemis/driver/xt_client.py

```diff
@@ -0,0 +1,344 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Created on 16 Aug 2019
+
+@author: Thera Pals
+
+Copyright © 2019 Thera Pals, Delmic
+
+This file is part of Delmic Acquisition Software.
+
+Delmic Acquisition Software is free software: you can redistribute it and/or modify it under the terms of the GNU
+General Public License as published by the Free Software Foundation, either version 2 of the License, or (at your
+option) any later version.
+
+Delmic Acquisition Software is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+more details.
+
+You should have received a copy of the GNU General Public License along with Delmic Acquisition Software. If not, see
+http://www.gnu.org/licenses/.
+"""
+from __future__ import division, print_function
+
+import logging
+
+import msgpack_numpy as mn
+import zerorpc
+
+mn.patch()
+
+# Necessary, otherwise nothing at all will be shown
+logging.basicConfig(format="%(asctime)s  %(levelname)-7s %(module)-15s: %(message)s")
+logging.getLogger().setLevel(logging.INFO)
+
+
+class MicroscopeClient:
+    """
+    Class to communicate with a Microscope server via the ZeroRPC protocol.
+    """
+
+    def __init__(self, server_address="tcp://192.168.1.1:4242", timeout=30):
+        """
+        Parameters
+        ----------
+        server_address: str
+            server address of the Microscope server as a string.
+        timeout: int
+            Time in seconds the client should wait for a response from the server.
+        """
+        # set heartbeat to None on client and server side, otherwise after two missed heartbeats the client thinks the
+        # connection is lost. A heartbeat happens every 5 seconds, when a function takes longer than 5 seconds to
+        # respond a heartbeat is skipped. timeout controls how long a call can take to respond, the default is 30
+        # seconds.
+        self.client = zerorpc.Client(server_address, heartbeat=None, timeout=timeout)
+
+    def list_available_channels(self):
+        """List all available channels and their current state as a dict."""
+        available_channels = self.client.list_available_channels()
+        return available_channels
+
+    def move_stage(self, position, rel=False):
+        """
+        Move the stage the given position in meter. This is non-blocking.
+
+        Parameters
+        ----------
+        position: dict(string->float)
+            Absolute or relative position to move the stage to per axes in m. Axes are 'x' and 'y'.
+        rel: boolean
+            If True the staged is moved relative to the current position of the stage, by the distance specified in
+            position. If false the stage is moved to the absolute position.
+        """
+        self.client.move_stage(position, rel)
```

Comment:
How are positions handled, which are out of range? Maybe add info about this in doc string.